File: pdlearning/src/microservices/learner-microservice/Api/Application/Models/MyAssignmentModel.cs
using System;
using Microservice.Learner.Domain.Entities;
using Microservice.Learner.Domain.ValueObject;

namespace Microservice.Learner.Application.Models
{
    public class MyAssignmentModel
    {
        public MyAssignmentModel(MyAssignment entity)
        {
            ParticipantAssignmentTrackId = entity.Id;
            RegistrationId = entity.RegistrationId;
            UserId = entity.UserId;
            AssignmentId = entity.AssignmentId;
            Status = entity.Status;
            SubmittedDate = entity.SubmittedDate;
            StartDate = entity.StartDate;
            EndDate = entity.EndDate;
        }

        public Guid ParticipantAssignmentTrackId { get; set; }

        public Guid RegistrationId { get; set; }

        public Guid UserId { get; set; }

        public Guid AssignmentId { get; set; }

        public MyAssignmentStatus Status { get; set; }

        public DateTime? SubmittedDate { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/learner/models/my-assignment.model.ts
export interface IMyAssignment {
  participantAssignmentTrackId: string;
  registrationId: string;
  userId: string;
  assignmentId: string;
  status: MyAssignmentStatus;
  submittedDate?: Date;
  commentNotSeenIds?: string[];
  startDate?: Date;
  endDate?: Date;
}

export class MyAssignment implements IMyAssignment {
  public participantAssignmentTrackId: string;
  public registrationId: string;
  public userId: string;
  public assignmentId: string;
  public status: MyAssignmentStatus;
  public submittedDate?: Date;
  public commentNotSeenIds?: string[];
  public startDate?: Date = new Date();
  public endDate?: Date = new Date();
  constructor(data?: IMyAssignment) {
    if (!data) {
      return;
    }
    this.participantAssignmentTrackId = data.participantAssignmentTrackId;
    this.registrationId = data.registrationId;
    this.userId = data.userId;
    this.assignmentId = data.assignmentId;
    this.status = data.status;
    this.submittedDate = data.submittedDate ? new Date(data.submittedDate) : undefined;
    this.commentNotSeenIds = data.commentNotSeenIds;
    this.startDate = data.startDate ? new Date(data.startDate) : undefined;
    this.endDate = data.endDate ? new Date(data.endDate) : undefined;
  }
}

export enum MyAssignmentStatus {
  NotStarted = 'NotStarted',
  InProgress = 'InProgress',
  Completed = 'Completed',
  Incomplete = 'Incomplete',
  IncompletePendingSubmission = 'IncompletePendingSubmission',
  LateSubmission = 'LateSubmission'
}
